### PR TITLE
Filtering automation logs and adding consistent prefix to the error logs

### PR DESCRIPTION
**Pull request by raych1**
**Description**
- filtered the logs to be rendered in the spec PR comment
- added the consistent prefix for the error logs

**Screenshot of the testing PR**
https://github.com/test-repo-billy/azure-rest-api-specs/pull/3391#issuecomment-2202365833
image

### CODE REVIEW

File: eng/mgmt/automation/swagger_to_sdk_config.json [hunk 1]
```diff
@@ -10,7 +10,12 @@
   },
   "generateOptions": {
     "generateScript": {
-      "path": "./eng/mgmt/automation/generate.py"
+      "path": "./eng/mgmt/automation/generate.py",
+      "logPrefix": "Java-Sdk-Automation",
+      "stderr": {
+        "showInComment": "(\\[GENERATE\\]\\[Error\\]|\\[COMPILE\\] Maven build fail|ExternalError).*",
```
Comment (XiaofeiCao) [inline eng/mgmt/automation/swagger_to_sdk_config.json:16]: When do we use `ExternalError`?

**Review by XiaofeiCao — APPROVED** [review]

File: eng/mgmt/automation/swagger_to_sdk_config.json (hunk 1 above)
Comment (raych1) [reply eng/mgmt/automation/swagger_to_sdk_config.json:16]: tsp-client would log this when tsp compiler/emitter fails.

Comment (XiaofeiCao) [reply eng/mgmt/automation/swagger_to_sdk_config.json:16]: I see.

**Review by weidongxu-microsoft — APPROVED** [review]
Consideration for future work is to directly extract from log of error level.

The log pattern is here
https://github.com/test-repo-billy/azure-sdk-for-java/blob/f80a844e3c27d7064d3357134fa19c35eaf0d138/eng/mgmt/automation/generate.py#L394

**Pull request merged**